src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,543 @@
+ */
+function parseSignature(textRaw, values) {
+  /**
+   * @type {import('../types.d.ts').MethodSignature}
+   */
+  const signature = {
+    params: [],
+  };
+
+  const rawParameters = values.filter(value => {
+    if (value.name === 'return') {
+      signature.return = value;
+      return false;
+    }
+
+    return true;
+  });
+
+  /**
+   * Extract a list of the signatures from the method's declaration
+   * @example `[sources[, options]]`
+   */
+  let [, declaredParameters] = `\`${textRaw}\``.match(PARAM_EXPRESSION) || [];
+
+  if (!declaredParameters) {
+    signature.params = rawParameters;
+
+    return signature;
+  }
+
+  /**
+   * @type {string[]}
+   * @example ['sources[,', 'options]]']
+   */
+  declaredParameters = declaredParameters.split(',');
+
+  let optionalDepth = 0;
+  const optionalCharDict = { '[': 1, ' ': 0, ']': -1 };
+
+  declaredParameters.forEach((declaredParameter, i) => {
+    /**
+     * @example 'length]]'
+     * @example 'arrayBuffer['
+     * @example '[sources['
+     * @example 'end'
+     */
+    declaredParameter = declaredParameter.trim();
+
+    if (!declaredParameter) {
+      throw new Error(`Empty parameter slot: ${textRaw}`);
+    }
+
+    // We need to find out if this parameter is optional or not. We can tell this
+    //  if we're wrapped in brackets, so let's look for them.
+
+    let pos;
+    for (pos = 0; pos < declaredParameter.length; pos++) {
+      const levelChange = optionalCharDict[declaredParameter[pos]];
+
+      if (levelChange === undefined) {
+        break;
+      }
+
+      optionalDepth += levelChange;
+    }
+
+    // Cut off any trailing brackets
+    declaredParameter = declaredParameter.substring(pos);
+
+    const isParameterOptional = optionalDepth > 0;
+
+    for (pos = declaredParameter.length - 1; pos >= 0; pos--) {
+      const levelChange = optionalCharDict[declaredParameter[pos]];
+
+      if (levelChange === undefined) {
+        break;
+      }
+
+      optionalDepth += levelChange;
+    }
+
+    // Cut off any leading brackets
+    declaredParameter = declaredParameter.substring(0, pos + 1);
+
+    // Default value of this parameter in the method's declaration
+    let defaultValue;
+
+    const equalSignPos = declaredParameter.indexOf('=');
+    if (equalSignPos !== -1) {
+      // We have a default value, save it and then cut it off of the signature
+      defaultValue = declaredParameter.substring(equalSignPos, 1).trim();
+      declaredParameter = declaredParameter.substring(0, equalSignPos);
+    }
+
+    let parameter = rawParameters[i];
+    if (!parameter || declaredParameter !== parameter.name) {
+      // If we're here then the method likely has shared signatures
+      //  Something like, `new Console(stdout[, stderr][, ignoreErrors])` and
+      //  `new Console(options)`
+      parameter = undefined;
+
+      // Try finding a parameter this is being shared with
+      for (const otherParam of rawParameters) {
```

Comment:
This seems a bit convoluted. Do you believe there's something you can do here to optimise the code?

I can see a few ways of optimising the signature generation and reducing its footprint.